### ISSUE TITLE
API returns 400 for some requests due to "bad" Content-type header

### ISSUE THREAD

**Issue by agaffney** [StackStorm/st2#2504]
It seems that *some* API endpoints will return a HTTP 400 when presented with a header such as:

```
Content-type: application/json; charset=utf-8
```

but works fine with:

```
Content-type: application/json
```

I ran into this with a POST to the URL `/webhooks/st2`. I'm currently running version `1.4dev-52`

**Comment by manasdk** [comment]
Thanks for the report. Seems fixable :)

**Comment by manasdk** [comment]
Should be a simple enough fix https://github.com/StackStorm/st2/blob/master/st2api/st2api/controllers/v1/webhooks.py#L119

Care to send a PR our way?

**Comment by agaffney** [comment]
Unfortunately, since the problem is no longer standing in my way, I lack the motivation to put together a PR (right now). You're lucky you got the bug report out of me :)

**Comment by manasdk** [comment]
>  You're lucky you got the bug report out of me :)

We certainly appreciate the bug report.

**Comment by Kami** [comment]
@agaffney Good catch and thanks for the report.

I will add a test case and push a fix later today.

**Comment by Kami** [comment]
I've merged a fix (#2506) into master.

@agaffney You should be able to upgrade to the latest 1.4dev version which includes this fix once the packages are built.
